### PR TITLE
Bug 1717734: status: add related dnses to operator status

### PR DESCRIPTION
**Pull request by ironcladlou**
Add related dnses to operator status to ensure dnses are captured
by must-gather.

`oc adm must-gather` output:

```
$ tree cluster-scoped-resources/operator.openshift.io/dnses/
cluster-scoped-resources/operator.openshift.io/dnses/
└── default.yaml

0 directories, 1 file
```

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1717734

TODO
- [x] Merge https://github.com/openshift/cluster-dns-operator/pull/116

### CODE REVIEW

**Comment by ironcladlou** [comment]
Waiting on dependencies to merge
/hold

**Comment by ironcladlou** [comment]
@Miciah @danehans this is ready to review, thanks!

File: pkg/operator/controller/status.go [hunk 1]
```diff
@@ -61,6 +61,26 @@ func (r *reconciler) syncOperatorStatus() error {
 	if err != nil {
 		return fmt.Errorf("failed to get operator state: %v", err)
 	}
+
+	related := []configv1.ObjectReference{
+		{
+			Resource: "namespaces",
+			Name:     "openshift-dns-operator",
```
Comment (Miciah) [inline pkg/operator/controller/status.go]: Tech debt accounting note: This maybe should be `r.Namespace`.

**Pull request merged**